crates/rome_js_formatter/src/utils/mod.rs

```diff
@@ -649,3 +650,118 @@ pub(crate) fn is_call_like_expression(expression: &JsAnyExpression) -> bool {
             | JsAnyExpression::JsCallExpression(_)
     )
 }
+
+/// Data structure used to merge into one the following nodes:
+///
+/// - [JsAnyObjectMemberName]
+/// - [JsAnyClassMemberName]
+/// - [JsLiteralMemberName]
+///
+/// Once merged, the enum is used to get specific members (the literal ones) and elide
+/// the quotes from them, when the algorithm sees fit
+pub(crate) enum MemberName {
+    Object(JsAnyObjectMemberName),
+    Class(JsAnyClassMemberName),
+    Literal(JsLiteralMemberName),
+}
+
+impl From<JsAnyClassMemberName> for MemberName {
+    fn from(node: JsAnyClassMemberName) -> Self {
+        Self::Class(node)
+    }
+}
+
+impl From<JsAnyObjectMemberName> for MemberName {
+    fn from(node: JsAnyObjectMemberName) -> Self {
+        Self::Object(node)
+    }
+}
+
+impl From<JsLiteralMemberName> for MemberName {
+    fn from(literal: JsLiteralMemberName) -> Self {
+        Self::Literal(literal)
+    }
+}
+
+const QUOTES_TO_OMIT: [char; 2] = ['\"', '\''];
+
+pub(crate) enum MemberContext {
+    Type,
+    Member,
+}
+
+impl MemberContext {
+    /// We can change the text only if there alphanumeric or alphabetic characters, depending on the mode
+    fn text_can_be_replaced(&self, text_to_check: &str) -> bool {
+        // Text here is quoteless. If it's empty, it means it is an empty string and we can't
+        // do any transformation
+        if text_to_check.is_empty() {
+            return false;
+        }
+        // we split by the characters that can be considered valid identifiers
+        let mut patterns = text_to_check.split(['_', '$']);
+        match self {
+            MemberContext::Type => {
+                patterns.all(|sub_text| sub_text.chars().all(char::is_alphabetic))
+            }
+            MemberContext::Member => {
+                patterns.all(|sub_text| sub_text.chars().all(char::is_alphanumeric))
+            }
```

Comment:
Nit: can't this be written as

```suggestion
        match self {
            MemberContext::Type => {
                text.chars().all(|c| matches!(c, '_', '$') || c.is_alphabetic)
            }
            MemberContext::Member => {
                text.chars().all(|c| matches!(c, '_', '$') || c.is_alphanumeric)
            }
```

Reply:
Not sure, an identifier can be "_&£_foo", and this case we can't elide the quotes.

Would your suggestion cover  case?

Reply:
The problem here is that you can't remove the quotes because of £ and not because of the `_`. For example, prettier removes the quotes for `let a = {"_f_$$_oo": 1 }`